tests/federation_room_join_test.go

```diff
@@ -290,7 +290,9 @@ func TestCannotSendNonLeaveViaSendLeaveV2(t *testing.T) {
 // and checks that they are all rejected.
 func testValidationForSendMembershipEndpoint(t *testing.T, baseApiPath, expectedMembership string, createRoomOpts map[string]interface{}) {
 	if createRoomOpts == nil {
-		createRoomOpts = make(map[string]interface{})
+		createRoomOpts = map[string]interface{}{
+			"preset": "public_chat",
+		}
```

Comment:
we'll need to change https://github.com/matrix-org/complement/blob/bdf545bf03b78b442e539ac78192001037e29aa9/tests/msc2403_test.go#L476 too, I think?

Reply:
because that test sets an explicit createRoomOpts, so it will otherwise create a room with no preset (and hence that will suffer the same problem as you're trying to solve here.)

Reply:
Oh I see, yes of course. Sorry I thought you meant putting the room version in the create opts.